### DISCUSSION TITLE
Advance Module Manager tries to update database when rendering a module

### DISCUSSION THREAD

**Discussion by desbear** [joomlatools/joomlatools-pages#886] [Q&A] [answered]
Hi~

Before I succeeded in not setting the module in the joomla backend, and directly loading the flexicontent module on the pages. After adjusting my environment recently, the module cannot be displayed again. 

If I go to the backend of joomla to set the module at this position. The module  displayed successfully. On the other hand, the page I created, often had a 1062 error message when refreshing browser, so I turned on the debugger, error message, such as this pdf [https://a.cl.ly/qGu75XrE](https://a.cl.ly/qGu75XrE) 

Seems to be no  errors after I close the advance module manager. But the advance module manager of regularlabs is quite important to me, and I still need to turn it on. 

I'm not sure I use <ktml:module position="product-userstory" name="flexicontent" params="<?= json($params, true)?>"> but currently it doesn't show this module, if I don't enable advance module manager , which part should I check to identify the problem?

**Comment by johanjanssens** [comment, marked as answer]
The 1062 is a database related error. In this case it seems this is due to a conflict between the Advanced Module Manager and Falang. If you check the callstack you see that: `AdvancedModulesModelModule->saveAdvancedParams()` is being called and intercepted by Falang `JFalangDatabase->execute()` which isn't able to execute the intercepted query.

### What is happening

1. Advanced Module Manager  is trying to store params in the database
2. Falang is intercepting the query and trying to translate it

For some reason Advanced Module Manager is executing a update query for the module params, it decides this in `com_advancedmodules/models/module.php:1675` This is really bad practice, a frontend GET request should never trigger an update query. 
 
### Next steps
 
Check `com_advancedmodules/models/module.php:1675` and figure out what is causing the `$changed` to be `true`, probably it's related to the params you are passing in, then go from there. If you can make sure `$changed` remains false the query will not trigger and all will work.

**Reply by desbear** [reply]
I had uninstall falang and disable Advance Module Manager. But my code still not show the flexicontent module. 

Even if I don't use params in module, it can't be displayed simply through <ktml:module position="product-userstory" name="flexicontent" >(but it work with mod_menu), but if the module in this specific position is set directly in the backend, it can be displayed. 
Do you know if this could be something missing from my environment?

Currently I don't have any error messages on this page.
```php
`<?php
$params = [
    "layout" => "2022-userstory-list", 
    "count" => 10,
    "count_feat" => 0,
    "method_tags" => "3", 
    "tag_ids" => [ "15", "16", "322"], 
    "method_types" => "3", 
    "types" => ["2"],
    "cuttitle" => "40", 
    "use_fields" => "1",
    "fields" => "userimages,summary,companynameandposition,respondent,articleimage",
]
?>`
`<ktml:module position="product-userstory" name="flexicontent" params="<?= json($params, true)?>">`
```

**Reply by johanjanssens** [reply]
Happy to help, but this is a bit hard to go from here. I would need to get access to your exact setup and even then debugging this will take quite some time. 

Before go there, what is it that the Flexicontent module needs to display? Instead of using the module, might be easier to build it yourself as a partial in Pages. 

This would require you get the data directly from the Flexicontent database, which can be done quite easily with Pages. If you don't have much advanced needs (aka, joining data from diff db tables) the[ database collection](https://github.com/joomlatools/joomlatools-pages/wiki/Database-Collection) should do the trick.  It might however be a bit too basic for a Flexicontent use-case. In that case would need to create an Pages extension for Flexicontent, which is not that hard to do either.
